pkg/policy/selectorcache.go

```diff
@@ -223,12 +239,46 @@ func (sc *SelectorCache) GetModel() models.SelectorCache {
 	return selCacheMdl
 }
 
+func (sc *SelectorCache) handleUserNotifications() {
+	for {
+		sc.userMutex.Lock()
+		for len(sc.userNotes) == 0 {
+			sc.userCond.Wait()
```

Comment:
How come this doesn't deadlock? We are effectively holding `sc.userMutex.Lock()` so `queueUserNotification` would never be able to handle notifications because it won't be able to get a lock of `sc.userMutex`. I assume this could be fixed by [1]

Reply:
From the docs:

> Wait atomically unlocks c.L and suspends execution of the calling goroutine. After later resuming execution, Wait locks c.L before returning. Unlike in other systems, Wait cannot return unless awoken by Broadcast or Signal.

Because c.L is not locked when Wait first resumes, the caller typically cannot assume that the condition is true when Wait returns. Instead, the caller should Wait in a loop:

`userCond` is initialized as:
```
	sc.userCond = sync.NewCond(&sc.userMutex)
```
so `Wait()` will unlock `userMutex` and locks it again before returning. `userMutex` must be held to read `userNotes`.

Reply:
ahh I forgot about that minor detail, see how comments are nice? :smile: 
>// userMutex protects userNotes and is linked to userCond